Assets/Mirror/Runtime/NetworkIdentity.cs

```diff
@@ -217,6 +218,25 @@ bool ThisIsASceneObjectWithPrefabParent(out GameObject prefab)
             return true;
         }
 
+        static uint GetRandomSceneID()
+        {
+            // use Crypto RNG to avoid having time based IDs
+            using (RNGCryptoServiceProvider rng = new RNGCryptoServiceProvider())
+            {
+                byte[] bytes = new byte[4];
+                uint id = 0;
+                // exclude '0' because that's for unassigned sceneIDs
+                // regenerate in case of an existing ID
+                while (id == 0 || sceneIds.ContainsKey(id))
```

Comment:
please leave the while part out of this function for now.
it's very useful to have a function that simply generates a RNG uint and no magic :)